app/src/main/java/com/andrew/tasky/presentation/event_detail/EventDetailViewModel.kt

```diff
@@ -0,0 +1,129 @@
+    }
+
+    private val _selectedStartDate = MutableStateFlow(LocalDate.now())
+    val selectedStartDate = _selectedStartDate.asStateFlow()
+    fun setStartDate(selectedStartDate: LocalDate){
+        _selectedStartDate.value = selectedStartDate
+    }
+
+    private val _selectedStartTime = MutableStateFlow(LocalTime.now())
+    val selectedStartTime = _selectedStartTime.asStateFlow()
+    fun setStartTime(selectedStartTime: LocalTime){
+        _selectedStartTime.value = selectedStartTime
+    }
+
+    private val _selectedEndDate = MutableStateFlow(LocalDate.now())
+    val selectedEndDate = _selectedEndDate.asStateFlow()
+    fun setEndDate(selectedEndDate: LocalDate){
+        _selectedEndDate.value = selectedEndDate
+    }
+
+    private val _selectedEndTime = MutableStateFlow(LocalTime.now())
+    val selectedEndTime = _selectedEndTime.asStateFlow()
+    fun setEndTime(selectedEndTime: LocalTime){
+        _selectedEndTime.value = selectedEndTime
+    }
+
+    private val _selectedReminderTime = MutableStateFlow(ReminderTimes.TEN_MINUTES_BEFORE)
+    val selectedReminderTime = _selectedReminderTime.asStateFlow()
+    fun setSelectedReminderTime(selectedReminderTime: ReminderTimes){
+        _selectedReminderTime.value = selectedReminderTime
+    }
+
+    private val _photos = MutableStateFlow(listOf<Uri>())
+    val photos = _photos.asStateFlow()
+    fun addPhoto(uri: Uri){
+        _photos.value += uri
+    }
+    fun deletePhoto(index1: Int){
+        val updatedPhotos = photos.value.filterIndexed { index, _ -> index != index1  }
+        _photos.value = updatedPhotos
+    }
+    fun setupPhotos(photoList: List<Uri>){
+        _photos.value = photoList
+    }
+
+    private val _attendees = MutableStateFlow(listOf<Attendee>())
+    val attendees = _attendees.asStateFlow()
+    fun addAttendee(attendee: Attendee){
+        _attendees.value += attendee
+    }
+
+    val goingAttendees = attendees.map {
+        it.filter { attendee ->  attendee.isAttending }
+    }.stateIn(viewModelScope, SharingStarted.WhileSubscribed(5000), emptyList())
+
+    val notGoingAttendees = attendees.map {
+        it.filter { attendee ->  !attendee.isAttending }
+    }.stateIn(viewModelScope, SharingStarted.WhileSubscribed(5000), emptyList())
+
+    fun deleteAttendee(attendee: Attendee){
+        val updatedAttendees = attendees.value.filter { it != attendee }
+        _attendees.value = updatedAttendees
+    }
+
+
+    private val _selectedAttendeeButton = MutableStateFlow(ShowListOfAttendeesButtonTypes.ALL)
+    val selectedShowListOfAttendeesButton = _selectedAttendeeButton.asStateFlow()
+    fun showAllAttendees(){
+        _selectedAttendeeButton.value = ShowListOfAttendeesButtonTypes.ALL
+    }
+    fun showGoingAttendees(){
+        _selectedAttendeeButton.value = ShowListOfAttendeesButtonTypes.GOING
+    }
+    fun showNotGoingAttendees(){
+        _selectedAttendeeButton.value = ShowListOfAttendeesButtonTypes.NOT_GOING
+    }
```

Comment:
You could also just create one function for this (BTW, I'd also prefer `AttendeeFilterType` over `ShowListOfAttendeesButtonTypes`):
```
fun setAttendeeFilterType(type: ShowListOfAttendeesButtonTypes) {
    _selectedAttendeeButton.value = type
}
```